course_contents/2_intro_to_python/notes/02. Loops.md

```diff
@@ -32,50 +35,61 @@ while counter > 0: # the counter never gets changed, causing the conditional to
   print(f"{counter} seconds left until New Year!")
 ```
 
+**Extra resources**
+[While loops](https://www.teclado.com/30-days-of-python/python-30-day-8-while-loops#object-object-loops)
+
 ### The `for` loop
-- for loop runs a given number of times
-- general syntax for `for` loop is:
+- `for` loop runs a given number of times
+- general syntax for using the `for` loop is:
 
 ```py
 for variable in iterable:
-    block_of_code
+    <block_of_code>
 ```
+
 - an iterable is an object that can be iterated over e.g. a list
+> do we need the next line? (line 47)
 - an object is iterable if it can return the iterator object (more on that in the OOP section)
 
 An example of the `for` loop:
 
 ```py
 numbers = [1,2,3,4,5]
-for i in numbers:
-    print(i)
+for number in numbers:
+    print(number)
 ```
 
 **Extra resources**
-[How does the for loop work in Python?](https://python.tecladocode.com/3_continuous_countries/1_loops.html#how-does-the-for-loop-work-in-python)
+[Defining a for loop](https://www.teclado.com/30-days-of-python/python-30-day-6-for-loops#defining-a-object-object-loop)
 
-#### The `range()` method
-- the `range()` method returns a sequence of integers in a given range
-- arguments that are passed to the method are:
+#### The `range()` function
+> lines 65-68 can be changed by showing the syntax as we did before and then using the design to point out each of the arguments
+> we can do the same thing to explain lines 70-71
+- the `range()` function returns a sequence of integers in a given range
+- arguments that are passed to the function are:
     1. `start` - an integer marking the start of a sequence
     2. `stop` - an integer marking the end of a sequence. Important - `range()` returns integers up to `stop-1`
     3. `step` - an integer representing the size of the step i.e. the difference between consecutive integers in a sequence
 - the numbers are generated by starting from `start` and incrementing it by `step` until `stop-1` is reached
 - the only argument that has to be stated explicitly is the `stop`
 - if not specified, Python will implicitly set `start` to 0 and `step` to 1
 
-An example of the `range()` method used in a `for` loop:
-
+An example of the `range()` function used in a `for` loop:
+> this is quite a simple example, we can go for something more "real lifelike" if needed
 ```py
-for i in range(5): #the range() method can be used to specify number of iterations of a loop
-  print(i) #the numbers printed will be: 0,1,2,3,4
+for i in range(5):
+  print(i) 
```

Comment:
I agree with your comments here, I think showing the code in quite a large format and then putting callouts for each argument might be an easy way to show what each one does.

But it would also be nice to show:

- `range(stop)`
- `range(start, stop)`
- `range(start, stop, step)`

I don't think we need much for this, maybe just some variation of this:

```
range(5)  # [1, 2, 3, 4]
range(3, 6)  # [3, 4, 5]
range(2, 14, 3)  # [2, 5, 8, 12]
```

^ although for that, we need to do it in a way that doesn't suggest that `range()` returns a `list`.